src/store/account/mobileSagas.ts

```diff
@@ -52,6 +56,29 @@ function* watchAccountRecovery() {
   })
 }
 
+export function* setHasSignedInOnMobile(account: User) {
+  const isMobileUser = window.localStorage.getItem(IS_MOBILE_USER_KEY)
+  if (!isMobileUser || isMobileUser !== 'true') {
+    try {
+      // Legacy method to update whether a user has signed in on
+      // native mobile. Used in identity service for notification indexing
+      yield call(AudiusBackend.updateUserEvent, {
+        hasSignedInNativeMobile: true
+      })
+      // Updates the user metadata with an event `is_mobile_user` set to true
+      // if the account is being fetched from a mobile context
+      yield call(updateProfileAsync, {
```

Comment:
If sign out then in, I think we'll try to write out the is_mobile_user flag again? Maybe we could check for it here (i.e. if it's already true, do nothing)

Reply:
we need to clear it unfortunately b/c they could sign in as a different user. the alternative solution here would be to go through a lot of trouble of piping out is_mobile_user from the discovery node side and serve it on the current user request, but i think that's heavyweight